### ISSUE TITLE
Clarity needed in Start tasks concurrently

### ISSUE THREAD

**Issue by AxelDelsol** [dotnet/docs#20680]
[Enter feedback here]

In the section "Start tasks concurrently", no speed up is observed compared to the Sync version. 

I think it is because the Async static methods do not use await (there are actually 3 warnings telling us that is will run synchronously).

To have an actual speed up, one need to use await anytime there is a Task.Delay(xxx) but this part is not explained anywhere in the tutorial.

Otherwise, awesome tutorial.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 17d27a0d-c072-8c9f-7e31-3de60bee9ae8
* Version Independent ID: e0c09794-e712-19ff-cb9f-12648d60d4f6
* Content: [Asynchronous programming in C#](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/async/#await-tasks-efficiently)
* Content Source: [docs/csharp/programming-guide/concepts/async/index.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/concepts/async/index.md)
* Product: **dotnet-csharp**
* Technology: **csharp-async**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

**Comment by BillWagner** [comment]
I agree with your point @AxelDelsol I was concerned about this section.

My primary goal was to make one change at a time, introducing one new concept. That does mean some of the interim steps aren't production quality. I did try to explain that in the accompanying text. 

I'll add this to the backlog to update when I next update this.
